### PR TITLE
ci: move from manifest to config file

### PR DESCRIPTION
**Pull request by pendo324**
Issue #, if available: release-please action not triggering as expected, see [errors](https://github.com/runfinch/finch/actions/runs/7292396775/job/20104474662) and previous PR (https://github.com/runfinch/finch/pull/738)

*Description of changes:*
- I re-read the release-please-action [documentation](https://github.com/google-github-actions/release-please-action?tab=readme-ov-file#advanced-release-configuration), and I think I put the config in the wrong file in my previous PR.
- Followed the instructions [here](https://github.com/googleapis/release-please/blob/main/docs/manifest-releaser.md) to create a manifest release since that seems to be what release-please requires when using a config file on v4
- Moved release-type to the config file instead of the yaml directly because the action [seems to branch](https://github.com/google-github-actions/release-please-action/blob/cc61a07e2da466bebbc19b3a7dd01d6aecb20d1e/src/index.ts#L83) its behavior depending on if that's present or not
 
*Testing done:*
- Tested on my fork, but its hard to replicate the upstream perfectly

- [x] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: release-please-config.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
+    "$schema": "https://raw.githubusercontent.com/googleapis/release-please/main/schemas/config.json",
+    "release-type": "go",
+    "changelog-sections": [
+        {
+            "type": "build",
```
Comment (weikequ) [inline release-please-config.json:6]: We have 2 build types, one hidden and one not.

File: release-please-config.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            "section": "Code Refactoring",
+            "hidden": true
+        },
+        {
+            "type": "test",
+            "section": "Tests",
+            "hidden": true
+        },
+        {
+            "type": "build",
```
Comment (weikequ) [inline release-please-config.json:51]: See above

**Review by weikequ — COMMENTED** [review]
> I re-read the release-please-action [documentation](https://github.com/runfinch/finch/actions/runs/7292396775/job/20104474662), and I think I put the config in the wrong file in my previous PR.

This link doesn't point to the right place. Did you mean https://github.com/google-github-actions/release-please-action?tab=readme-ov-file#advanced-release-configuration ?

> Followed the instructions here to create a manifest release since that seems to be what release-please requires when using a config file on v4

No link for "here", did you mean https://github.com/googleapis/release-please/blob/main/docs/manifest-releaser.md ?

**Comment by pendo324** [comment]
> > I re-read the release-please-action [documentation](https://github.com/runfinch/finch/actions/runs/7292396775/job/20104474662), and I think I put the config in the wrong file in my previous PR.
> 
> This link doesn't point to the right place. Did you mean https://github.com/google-github-actions/release-please-action?tab=readme-ov-file#advanced-release-configuration ?
> 
> > Followed the instructions here to create a manifest release since that seems to be what release-please requires when using a config file on v4
> 
> No link for "here", did you mean https://github.com/googleapis/release-please/blob/main/docs/manifest-releaser.md ?

Yeah, it was a copy/paste error, fixed the links in the original PR, thanks.

File: .github/workflows/release-please.yaml [hunk 3]
```diff
@@ -9,5 +9,3 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: google-github-actions/release-please-action@v4
```
[the remark this answers is not in the document]
Comment (pendo324) [reply .github/workflows/release-please.yaml:11]: Seems like it defaults to secrets.GITHUB_TOKEN ([source](https://github.com/google-github-actions/release-please-action?tab=readme-ov-file#action-inputs)), which is not defined in this repo, but is created each time an action is run ([GH docs](https://github.com/google-github-actions/release-please-action?tab=readme-ov-file#action-inputs))

**Review by weikequ — APPROVED** [review]
Looks good, thanks!

**Pull request merged**